src/d3a/models/area/event_dispatcher.py

```diff
@@ -136,7 +136,7 @@ def create_agent_object(owner, higher_market, lower_market, is_spot_market):
                     return OneSidedAlternativePricingAgent(**agent_constructor_arguments)
                 else:
                     return OneSidedAgent(**agent_constructor_arguments)
-            elif ConstSettings.IAASettings.MARKET_TYPE == 2:
+            elif ConstSettings.IAASettings.MARKET_TYPE in [2, 4]:
```

Comment:
I think that it is a good opportunity to get rid of the two-sided pay as clear agent, since it is more or less the same as the two sided pay as bid agent